### PR TITLE
Move to Local-first data & unify connections

### PR DESCRIPTION
**Pull request by rlh1994**
1) Move to a local first data approach, I'm using csvs as they are easy to examine and alter, but end users may want to optimise the storage to parquet or pickle if they have large outputs? There is a new button on the dashboard to refresh the data which runs the queries, saves the files, and then reloads the dashboard
2) Unifies the warehouses to reduce duplication. The connector does all the work and a single variable makes the needed switch. Downside is the pipfile has to include all the connectors as I can't have 1 for bq and 1 for snowflake, but this is a small price to pay in my opinion.

A small bonus is it's now relatively easy to add new warehouses once the data is available there!

TODO:
- [x] Other folders
- [x] Update readme/pull instructions
- [x] Generate output data
  - [x] Fake where we don't provide input data
    - [x] all but consent
    - [x] consent
  - [x] Real where we provide input data
- [x] PRs for actual accelerators to use new format
- [x] Check consent works live in database (BQ and snowflake), no reason it shouldn't

### CODE REVIEW

**Review by emielver — COMMENTED** [review]
At a high level it looks good. Not sure if we need the `sf_bq` separately, would we want to rename it every time we add support for a new warehouse? And we have some duplication in `connect_bigquery` and `connect_snowflake`, is there a reason why they're two separate files instead of just a `connection.py` where you have an extra variable that is passed calls the right function when it's `snowflake` or `bigquery`?

**Comment by rlh1994** [comment]
> Not sure if we need the `sf_bq` separately, would we want to rename it every time we add support for a new warehouse?

I did this because in some we have databricks but those are notebooks. Maybe it should just be by Viz/reporting type and then we make clear what's supported within each?

> And we have some duplication in `connect_bigquery` and `connect_snowflake`, is there a reason why they're two separate files instead of just a `connection.py` where you have an extra variable that is passed calls the right function when it's `snowflake` or `bigquery`?

I had kept these separate as I planned to have 2 pip files so you didn't have to have both installed, but I agree, will merge them.

File: README.md [hunk 1]
```diff
@@ -5,16 +5,11 @@ This repo holds downloadable resources for use with [Snowplow Accelerators](http
...
-|       |    Streamlit
+|    Streamlit
+|    Looker
 
 E-Commerce Accelerator
-|----   Snowflake
-|       |   Looker
-|----   BigQuery
-|       |   Streamlit
+|   Looker
```
Comment (Jack-Keene) [inline README.md]: We'll need to change the instructions on downloading these repos in the actual accelerators as well. We can probably handle it in a similar way to the package versions in the template.

Comment (rlh1994) [reply README.md]: Yeah I'm working through getting it all standardised and having some example data if possible for each of them now, and then everything should actually require the same versions of packages and follow the same naming, so we'll have like 6 PRs to go live at the same time. 

Nice bonus of this is I'm pretty sure it adds bigquery for all the snowflake only ones, except we need the data upload for them still - if we go down the public dataset route that adds them completely!

**Comment by rlh1994** [comment]
I've also unifed the data upload parts for the enhanced analytics ones as it had some of it in some of them but not across all of them - not sure if it makes sense to remove these files from the accelerators directly as well @Jack-Keene ?

File: enhanced-consent-accelerator/streamlit/pages/1_CMP_Dashboard.py [hunk 2]
```diff
@@ -17,23 +26,27 @@
 c1, c2, c3 = st.columns((8, 1, 8))
 
 with c1:
-    fig = px.scatter(df_cmp_stats, x='CMP load time', y='First consent event type', width=700, height=400, title="<b> CMP load time vs consent choice<b>")
+    data['cmp_stats'] = data['cmp_stats'].rename(columns = {'cmp_load_time' : 'CMP load time',
+                                                            'first_consent_event_type': 'First consent event type',
+                                                            'cmp_interaction_time': 'CMP interaction time'
+                                                            })
+    fig = px.scatter(data['cmp_stats'], x='CMP load time', y='First consent event type', width=700, height=400, title="<b> CMP load time vs consent choice<b>")
```
Comment (Jack-Keene) [inline enhanced-consent-accelerator/streamlit/pages/1_CMP_Dashboard.py:33]:
```suggestion
    fig = px.scatter(data['cmp_stats'], x='cmp load time', y='first consent event type', width=700, height=400, title="<b> CMP load time vs consent choice<b>")
```
Not sure why, but it didn't work with capitalised x & y

File: enhanced-consent-accelerator/streamlit/pages/1_CMP_Dashboard.py [hunk 3]
```diff
@@ -17,23 +26,27 @@
...
     fig.update_layout(
         margin=dict(l=20, r=20, t=80, b=40),
     )
     st.plotly_chart(fig)
 
 with c3:
-    fig = px.scatter(df_cmp_stats, x='CMP interaction time', y='First consent event type', width=700, height=400, title="<b> CMP interaction time vs consent choice<b>")
+    fig = px.scatter(data['cmp_stats'], x='CMP interaction time', y='First consent event type', width=700, height=400, title="<b> CMP interaction time vs consent choice<b>")
```
Comment (Jack-Keene) [inline enhanced-consent-accelerator/streamlit/pages/1_CMP_Dashboard.py:40]:
```suggestion
    fig = px.scatter(data['cmp_stats'], x='cmp interaction time', y='first consent event type', width=700, height=400, title="<b> CMP interaction time vs consent choice<b>")
```

Same here

**Review by Jack-Keene — CHANGES_REQUESTED** [review]
LGTM apart from a few minor changes!

File: enhanced-consent-accelerator/streamlit/pages/1_CMP_Dashboard.py (hunk 2 above)
Comment (rlh1994) [reply enhanced-consent-accelerator/streamlit/pages/1_CMP_Dashboard.py:33]: My bad, I had saved an older version of the data with column names with spaces in (which bigquery does not allow), it should work now?

Comment (Jack-Keene) [reply enhanced-consent-accelerator/streamlit/pages/1_CMP_Dashboard.py:33]: I'm using snowflake, would BQ still break it? It just tried it again and still wasn't working. Same error as before

Comment (rlh1994) [reply enhanced-consent-accelerator/streamlit/pages/1_CMP_Dashboard.py:33]: I didn't push my commit, will do it in the morning 😅

**Pull request merged**